typing/ctype.mli

```diff
@@ -278,9 +279,12 @@ val unify_var: Env.t -> type_expr -> type_expr -> unit
         (* Same as [unify], but allow free univars when first type
            is a variable. *)
 val unify_alloc_mode: alloc_mode -> alloc_mode -> unit
-val filter_arrow: Env.t -> type_expr -> arg_label ->
+val filter_arrow: Env.t -> type_expr -> arg_label -> force_poly:bool ->
                   alloc_mode * type_expr * alloc_mode * type_expr
         (* A special case of unification (with l:'a -> 'b). *)
+val filter_mono: type_expr -> type_expr
```

Comment:
I think all the `filter` functions are equivalent to a particular call to `unify`. As you say, this one is equivalent to unifying with `Tpoly(newvar (), [])` and returning the result, so it is the same as these other functions.

I think these functions mostly exist for efficiency and convenience rather than using `unify` directly.

I've add documentation to the function to match the similar doc comments on the surrounding functions. I've also changed the implementation to `assert false` if the node isn't a `Tpoly` node at all because I think that makes it's use clearer.